test/integ_tests/test_reservation_arn.py

```diff
@@ -56,10 +56,13 @@ def test_create_task_via_reservation_arn_on_simulator(reservation_arn):
         )
 
 
-def test_create_job_via_invalid_reservation_arn_on_qpu(aws_session, reservation_arn):
+def test_create_job_via_invalid_reservation_arn_on_qpu(
+    aws_session, reservation_arn, braket_devices
+):
     with pytest.raises(ClientError, match="Reservation arn is invalid"):
+        online_dev = next(dev for dev in braket_devices if dev.is_available and dev.type == "QPU")
```

Comment:
What if no device is available? I think we should just remove this test.

Reply:
I figured either we have this test as a best effort or don't have it at all. I am not inclined to mock the test out so if you are fine with it being removed, I can cut that out.